format/ScanRequest.fbs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
This directory includes files that are meant to reflect the column format (e.g. types that all implementations are standardized on).  I'm not sure if your intention is to make the scan request a part of the specification or not.

However, I think this was probably placed here so it would get picked up by thrift.  I'm not sure what a good approach to take would be and I'll defer to future reviewers here.

Reply:
Okay, let's wait. In general, the intention is to add a generic flatbuffer schema that can serialize some scan request parameters to be sent through the network.

Reply:
So here is my current understanding.  Let me know if this seems off.  There are two pieces to this.

There is a ceph object class (called Skyhook?) which processes scan tasks and lives in a "contrib" directory.

There is a fragment / file format for Arrow that understands how to send scan requests to a ceph storage server in the skyhook format.

These two components aren't tightly coupled.  The only source of agreement is the Arrow columnar format and this flatbuffers file.  So for example (these are thought exercises, not things that will necessarily ever happen):

* Ceph could be running an older version of Skyhook built with Arrow version X and the dataset client could be running a newer version of Arrow version X+N.
* Skyhook could switch to some other library entirely in the future and as long as it continued to respect the flatbuffers format it would continue to work.
* A different non-arrow library (or an Arrow implementation in a different language) could decide to start sending requests to Skyhook and as long as they agreed upon the flatbuffers and arrow columnar format everything would continue to work.

Given the above I think the proper place for this flatbuffers file to live is in the same directory as the ceph object class.  This flatbuffers file is the API for skyhook.

Then, for building everything, the make files for that directory could produce two artifacts: A ceph object class and a small C++ "client library" which is just the output of the flatbuffers compiler.

Or you could skip the "client library" step and add an extra build step for the datasets module which runs the flatbuffers compiler.